ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/BlockFetch.hs

```diff
@@ -0,0 +1,141 @@
+
+import           Control.Monad (void)
+import           Control.Monad.Class.MonadTime
+import           Control.Tracer (nullTracer)
+import           Data.Hashable (Hashable)
+import           Data.Map.Strict (Map)
+import           Network.TypedProtocol.Channel (createConnectedChannels)
+import           Network.TypedProtocol.Driver.Simple
+                     (runConnectedPeersPipelined)
+import           Ouroboros.Consensus.Block (HasHeader)
+import           Ouroboros.Consensus.Block.Abstract (Header, Point (..))
+import qualified Ouroboros.Consensus.MiniProtocol.BlockFetch.ClientInterface as BlockFetchClientInterface
+import           Ouroboros.Consensus.Node.ProtocolInfo
+                     (NumCoreNodes (NumCoreNodes))
+import           Ouroboros.Consensus.Storage.ChainDB.API
+import           Ouroboros.Consensus.Util.IOLike (IOLike, STM, atomically,
+                     retry)
+import           Ouroboros.Consensus.Util.ResourceRegistry
+import           Ouroboros.Network.AnchoredFragment (AnchoredFragment)
+import qualified Ouroboros.Network.AnchoredFragment as AF
+import           Ouroboros.Network.BlockFetch (BlockFetchConfiguration (..),
+                     FetchClientRegistry, FetchMode (..), blockFetchLogic,
+                     bracketFetchClient, bracketKeepAliveClient)
+import           Ouroboros.Network.BlockFetch.Client (blockFetchClient)
+import           Ouroboros.Network.ControlMessage (ControlMessageSTM)
+import           Ouroboros.Network.NodeToNode.Version (NodeToNodeVersion,
+                     isPipeliningEnabled)
+import           Ouroboros.Network.Protocol.BlockFetch.Codec (codecBlockFetchId)
+import           Ouroboros.Network.Protocol.BlockFetch.Server
+                     (BlockFetchBlockSender (SendMsgNoBlocks, SendMsgStartBatch),
+                     BlockFetchSendBlocks (SendMsgBatchDone, SendMsgBlock),
+                     BlockFetchServer (..), blockFetchServerPeer)
+import           Ouroboros.Network.Protocol.BlockFetch.Type (ChainRange (..))
+import           Test.Util.Orphans.IOLike ()
+import           Test.Util.TestBlock (BlockConfig (TestBlockConfig), TestBlock)
+import           Test.Util.Time (dawnOfTime)
+
+
+startBlockFetchLogic
+  :: forall m peer.
+     (Hashable peer, Ord peer, IOLike m)
+  => ResourceRegistry m
+  -> ChainDB m TestBlock
+  -> FetchClientRegistry peer (Header TestBlock) TestBlock m
+  -> STM m (Map peer (AnchoredFragment (Header TestBlock)))
+  -> m ()
+startBlockFetchLogic registry chainDb fetchClientRegistry getCandidates = do
+    let slotForgeTime :: BlockFetchClientInterface.SlotForgeTimeOracle m TestBlock
+        slotForgeTime _ = pure dawnOfTime
+
+        blockFetchConsensusInterface =
+          BlockFetchClientInterface.mkBlockFetchConsensusInterface
+            (TestBlockConfig $ NumCoreNodes 0) -- Only needed when minting blocks
+            (BlockFetchClientInterface.defaultChainDbView chainDb)
+            getCandidates
+            (\_hdr -> 1000)
```

Comment:
There are quite some numeric constants in this function. Would it make sense to explain why we chose these values?

Reply:
```suggestion
            -- The size of headers in bytes is irrelevant because our tests
            -- do not serialize the blocks.
            (\_hdr -> 1000)
```